lib/spack/spack/llnl/util/filesystem.py

```diff
@@ -2331,7 +2331,7 @@ def find_system_libraries(libraries, shared=True):
         message = message.format(find_system_libraries.__name__, type(libraries))
         raise TypeError(message)
 
-    libraries_found = []
+    libraries_found = LibraryList([])
```

Comment:
NOTE: this was bug. This function returned a list instead of the promised `LibraryList`